pkg/services/accesscontrol/middleware.go

```diff
@@ -84,6 +84,7 @@ func deny(c *models.ReqContext, evaluator Evaluator, err error) {
 
 	if !c.IsApiRequest() {
 		// TODO(emil): I'd like to show a message after this redirect, not sure how that can be done?
+		writeRedirectCookie(c)
```

Comment:
This is not the same function, since this lies in a different package, is that intentional?

Reply:
@sakjur sorry if this is a misunderstanding, but would calling this not call the function here https://github.com/grafana/grafana/blob/main/pkg/services/accesscontrol/middleware.go#L130 ?

Reply:
It would, but you've modified the function that was copied from in `pkg/middleware/auth.go` rather than that function? It's possible that's what you want, but I'm asking about it since you're modifying `pkg/middleware.writeRedirectCookie` and adding a dependency on `pkg/services/accesscontrol.writeRedirectCookie` in the same PR and I want to double-check so that that's right.

Reply:
Yes, for this bug both the access control middeware and normal Auth middleware kick in. The AC middleware is what   gets accessed for authorization, but then redirects to the root directory. The normal auth middleware then kicks in to that route, and sets the cookie to just the /

Reply:
Ah, I think I understand.

Looks good then, let me just take a quick look and approve :+1: